frame/staking/src/lib.rs

```diff
@@ -3415,6 +880,16 @@ where
 	}
 }
 
+/// A `Convert` implementation that finds the stash of the given controller account,
+/// if any.
+pub struct StashOf<T>(sp_std::marker::PhantomData<T>);
```

Comment:
`types.rs`?

Reply:
Yeah I was thinking something like that for a follow up - we can  have `crate::pallet::types` and `crate::types` for ones we may want to expose publicly.